base/REPL.jl

```diff
@@ -323,11 +329,15 @@ end
 function hist_from_file(hp, file)
     hp.history_file = file
     seek(file, 0)
+    countlines = 0
     while true
+        local ch::Char
         mode = :julia
         line = hist_getline(file)
         isempty(line) && break
-        line[1] == '#' || error(invalid_history_message)
+        countlines += 1
+        line[1] != '#' &&
+            error(invalid_history_message, hex(line[1]), " at line ", countlines)
```

Comment:
Ah, simply because I hadn't used `repr` ever before!  (Still learning things about julia!)  Thanks, I'll change that as soon as I get home.